src/algebra/field.lean

```diff
@@ -5,6 +5,39 @@ Authors: Robert Lewis, Leonardo de Moura, Johannes Hölzl, Mario Carneiro
 -/
 import algebra.ring.basic
 import algebra.group_with_zero
+
+/-!
+# Fields and division rings
+
+This file introduces fields and division rings (also known as skewfields) and proves some basic
+statements about them. For a more extensive theory of fields, see `field_theory`.
+
+## Main definitions
+
+* `division_ring`: introduces the notion of a division ring as a `ring` such that `0 ≠ 1` and
+  `a * a⁻¹ = 1` for `a ≠ 0`
+* `field`: a division ring which is also a commutative ring.
+* `is_field`: a predicate on a ring that it is a field, i.e. that the multiplication is commutative,
+  that it has more than one element and that all non-zero elements have a multiplicative inverse.
+  In contrast to `field`, which contains the data of a function associating to an element of the
+  field its multiplicative inverse, this predicate only assumes the existence and can therefore more
+  easily be used to e.g. transfer along ring isomorphisms.
+
+## Implementation details
+
+By convention `0⁻¹ = 0` in a field or division ring. This is due to the fact that working with total
+functions has the advantage of not constantly having to check that `x ≠ 0` when writing `x⁻¹`. With
+this convention in place, some statements like `(a + b) * c⁻¹ = a * c⁻¹ + b * c⁻¹` still remain
+true, while others like the defining property `a * a⁻¹ = 1` need the assumption `a ≠ 0`. If you are
+a beginner in using Lean and are confused by that, you can read more about why this convention is
+taken in the Kevin Buzzard's
+[blogpost](https://xenaproject.wordpress.com/2020/07/05/division-by-zero-in-type-theory-a-faq/)
+
```

Comment:
`group_with_zero` is not commonly used in mathematics, so let's point users to that definition too.
```suggestion

A division ring or field is an example of a `group_with_zero`. If you cannot find
a division ring / field lemma that does not involve `+`, you can try looking for
a `group_with_zero` lemma instead.

```